crate/server/src/core/cover_crypt/update_policy.rs

```diff
@@ -138,6 +170,123 @@ fn rotate_policy(
+/// and return it
+fn clear_old_rotations_policy(
+    private_key: &Object,
+    cover_crypt_policy_attributes_to_update: &[Attribute],
+) -> KResult<Policy> {
+    // Recover the current policy
+    let private_key_attributes = private_key.attributes()?;
+    let mut policy = policy_from_attributes(private_key_attributes)?;
+
+    // Rotate the Attributes values in the Policy
+    for attr in cover_crypt_policy_attributes_to_update {
+        policy.clear_old_rotations(attr).map_err(|e| {
+            KmsError::KmipError(
+                ErrorReason::Unsupported_Cryptographic_Parameters,
+                e.to_string(),
+            )
+        })?;
+    }
+    trace!("The new policy is : {policy:#?}");
+    Ok(policy)
+}
+
+fn remove_policy_attributes(
+    private_key: &Object,
+    cover_crypt_policy_attributes_to_update: &[Attribute],
+) -> KResult<Policy> {
+    // Recover the current policy
+    let private_key_attributes = private_key.attributes()?;
+    let mut policy = policy_from_attributes(private_key_attributes)?;
+
+    // Remove the Attributes in the Policy
+    for attr in cover_crypt_policy_attributes_to_update {
+        policy.remove_attribute(attr.clone()).map_err(|e| {
+            KmsError::KmipError(
+                ErrorReason::Unsupported_Cryptographic_Parameters,
+                e.to_string(),
+            )
+        })?;
+    }
+    trace!("The new policy is : {policy:#?}");
+    Ok(policy)
+}
+
+fn disable_policy_attributes(
+    private_key: &Object,
+    cover_crypt_policy_attributes_to_update: &[Attribute],
+) -> KResult<Policy> {
+    // Recover the current policy
+    let private_key_attributes = private_key.attributes()?;
+    let mut policy = policy_from_attributes(private_key_attributes)?;
+
+    // Disable the Attributes in the Policy
+    for attr in cover_crypt_policy_attributes_to_update {
+        policy.disable_attribute(attr.clone()).map_err(|e| {
+            KmsError::KmipError(
+                ErrorReason::Unsupported_Cryptographic_Parameters,
+                e.to_string(),
+            )
+        })?;
+    }
+    trace!("The new policy is : {policy:#?}");
+    Ok(policy)
+}
+
+fn add_policy_attribute(
+    private_key: &Object,
+    cover_crypt_policy_attributes_to_update: &[Attribute],
+    encryption_hint: EncryptionHint,
+) -> KResult<Policy> {
+    // Recover the current policy
+    let private_key_attributes = private_key.attributes()?;
+    let mut policy = policy_from_attributes(private_key_attributes)?;
+
+    // Add the Attributes in the Policy
+    for attr in cover_crypt_policy_attributes_to_update {
+        policy
+            .add_attribute(attr.clone(), encryption_hint)
+            .map_err(|e| {
+                KmsError::KmipError(
+                    ErrorReason::Unsupported_Cryptographic_Parameters,
+                    e.to_string(),
+                )
+            })?;
+    }
+    trace!("The new policy is : {policy:#?}");
+    Ok(policy)
+}
```

Comment:
Here you need ownership on the input attributes so that you can add them to the policy.
```suggestion
fn add_policy_attribute(
    private_key: &Object,
    additions: Vec<(Attribute, EncryptionHint)>,
) -> KResult<Policy> {
    let private_key_attributes = private_key.attributes()?;
    let mut policy = policy_from_attributes(private_key_attributes)?;

    for (new_attribute, encryption_hint) in additions {
        policy
            .add_attribute(new_attribute, encryption_hint)
            .map_err(|e| {
                KmsError::KmipError(
                    ErrorReason::Unsupported_Cryptographic_Parameters,
                    e.to_string(),
                )
            })?;
    }
    trace!("The new policy is : {policy:#?}");
    Ok(policy)
}
```